src/pydap/model.py

```diff
@@ -333,6 +333,9 @@ def keys(self):
         """Method to emulate a dictionary, returning keys."""
         return self._keys[:]
 
+    def __len__(self):
```

Comment:
I've made it into a ``MutableMapping`` because this is what it is. Sounds good to you?